Raise exception when first_name is Fail

pkarman opened the pull request:
**Why**: Allow for testing when vendor clients raise exceptions.
An exception might happen e.g. when an upstream vendor returns a
non-200 response or is unreachable.

zachmargolis commented on .rubocop.yml:
```diff
@@ -12,6 +12,13 @@ AllCops:
     - 'proofer.gemspec'
 
+Metrics/BlockLength:
+  CountComments: false  # count full line comments?
```
extra nit: two spaces before the comment?

pkarman replied:
heh. that must be in the idp version as well since I copy/pasted.

zachmargolis commented on lib/proofer/vendor/mock.rb:
```diff
@@ -130,6 +135,10 @@ def fail_confirmation_with_bad_phone(session_id)
       end
 
+      def fail_resolution_with_exception(_uuid)
+        raise 'Failed to contact proofing vendor'
```
do we want to raise a particular kind of error? or just RuntimError like this?

pkarman replied:
I wondered that too. Ambivalent. Didn't really want to introduce a specific exception class just for this, since vendor-specific drivers could conceivably throw any kind of exception.

zachmargolis replied:
Makes sense, thanks!

zachmargolis commented:
LGTM +/- some comments

Commit:
Raise exception when first_name is Fail

**Why**: Allow for testing when vendor clients raise exceptions.
An exception might happen e.g. when an upstream vendor returns a
non-200 response or is unreachable.
